django/cantusdb_project/main_app/views/chant.py

```diff
@@ -776,56 +798,76 @@ def get_queryset(self) -> QuerySet:
                 # as a substring
                 feasts = Feast.objects.filter(name__icontains=feast)
                 q_obj_filter &= Q(feast__in=feasts)
-            if self.request.GET.get('order'):
-                if self.request.GET.get('order') == 'siglum':
-                    order = 'siglum'
-                elif self.request.GET.get('order') == 'incipit':
+            order: str = 'siglum'
+            order_get_param: Optional[str] = self.request.GET.get('order')
+            sort_get_param: Optional[str] = self.request.GET.get('sort')
+            if order_get_param is not None:
+                if order_get_param == 'siglum':
```

Comment:
You don't need this clause. (`pass` is, to me, code smell. It's useful while developing, but really shouldn't make its way into production code except in very limited circumstances.)

Reply:
for this and `if sort_get_param == "asc"` below, I think I was following the "explicit is better than implicit" dictum - I thought I was being helpful for future developers who will see a `?order=siglum` in the links on the template, try to find where this case is treated in the view, and would otherwise get confused when they can't find it. You have a better sense of this than I do, though, so happy to defer to you if you think these no-ops should just be skipped.

Alternatively, would skipping the initial `order: str = 'siglum'`, and setting things up along the lines of
```python
if order_get_param == 'a bunch of options here':
    order = 'one of the many options'
#...
elif order_get_param == 'siglum':
    order = 'siglum'
else:
    order = 'siglum'
```
work better?

Reply:
You set `order` above anyway, so it's `"siglum"` by default unless it gets changed.

I like the `else:` better, though, since it's more explicit about behaviour.

Reply:
if `order_get_param` doesn't match anything else in the `if` statement, it's `"siglum"`. So:

```
elif order_get_param == 'siglum':
    order = 'siglum'
else:
    order = 'siglum'
```

is redundant. You don't need the first case, since it would match on the `else` by default anyway.

Reply:
>  I thought I was being helpful for future developers who will see a ?order=siglum in the links on the template, try to find where this case is treated in the view, and would otherwise get confused when they can't find it. 

My sense is that this can be captured in a comment, rather than needing to capture it in code. Something like:

"# sort values: Allows 'asc' or 'desc'. Default if not specified is 'asc'."

Reply:
Fair enough - I'll make this change in both places. I like the idea of self-documenting code, but occasionally go further than necessary with it evidently. Thanks!